app/src/main/java/com/weatherxm/ui/Navigator.kt

```diff
@@ -525,14 +528,14 @@ class Navigator(private val analytics: AnalyticsWrapper) {
     fun showPhotoUpload(
         context: Context?,
         device: UIDevice,
-        photos: ArrayList<String>
+        photos: ArrayList<StationPhoto>
     ) {
         context?.let {
             it.startActivity(
                 Intent(it, PhotoUploadActivity::class.java)
                     .addFlags(Intent.FLAG_ACTIVITY_SINGLE_TOP or Intent.FLAG_ACTIVITY_CLEAR_TOP)
                     .putExtra(ARG_DEVICE, device)
-                    .putStringArrayListExtra(ARG_PHOTOS, photos)
+                    .putParcelableArrayListExtra(ARG_PHOTOS, photos)
```

Comment:
Instead of that, let's use the extension function found in `Views.kt` called `putParcelableList` , so that the `photos: ArrayList<StationPhoto>` param, will be `photos: List<StationPhoto>` and the `putParcelableList` will handle everything else by itself.

And you can remove the `ArrayList` from the `ArrayList(model.photos.filter { it.isLocal })` as we will pass a `List` now.